Source/ASDisplayNode.mm

```diff
@@ -2331,6 +2339,11 @@ - (void)_removeFromSupernodeIfEqualTo:(ASDisplayNode *)supernode
 
 - (void)_removeFromSupernode:(ASDisplayNode *)supernode view:(UIView *)view layer:(CALayer *)layer
 {
+  // TODO: Should we simply return early if supernode is nil? This currently gets called for every insertSubnode.
```

Comment:
The answer to this is probably no, because it is possible to have the _view / _layer be in a parent without having a supernode. I've fixed some bugs related to that before: specifically addSubnode: category exists on UIView, but when you perform that and there's no associated node, then removeFromSupernode did not remove it from the view (so it had assymetric behavior).

In that sense I think this is correct, so you can remove the TODO and optionally comment as to why we do this.